### PR TITLE
Files regarding Baklava update to CR8

### PR DESCRIPTION
**Pull request by martinvol**
### Description

Baklava was stuck in CR3 because it wasn't able to pass governance proposals. One that issue was fixed, a governance proposal was passed to speed run it to CR8 in one go. This is unconventional, as in mainnet Contract Releases go 1-by-1. Thus this will be one of the few proposals on Baklava that are unique, as they didn't pass in any other testnet.

To generate this file, the standard tooling described here was [used](https://docs.celo.org/community/release-process/smart-contracts), but the versions where set accordingly (rather than N+1, using the notation of the documentation). The tooling didn't have any issue processing this.

Note that ExchangeBRL.sol has different arguments than when ExchangeBRL was originally deploy on tesnets and mainnet, that is because this [PR](https://github.com/celo-org/celo-monorepo/pull/8334) was deployed in-between.

Version diff reports and release data are committed in this PR for future reference.

### Other changes

N/A

### Tested

In Baklava iteself.

### Related issues

- Fixes #[9946](https://github.com/celo-org/celo-monorepo/issues/9946)

### Backwards compatibility

None

### Documentation

The description of this PR itself.

### CODE REVIEW

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 1]
```diff
@@ -0,0 +1,859 @@
...
+            {
+              "contract": "Escrow",
+              "dependency": "Signatures",
+              "type": "LibraryLinkingChange"
+            }
+          ]
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+3",
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: On `core-contracts.v8`, Escrow appears to have version 1.2.0.0, why the +3 major increase?

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 2]
```diff
@@ -0,0 +1,859 @@
...
+            {
+              "contract": "FederatedAttestations",
+              "dependency": "Signatures",
+              "type": "LibraryLinkingChange"
+            }
+          ]
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: Does this mean FA is now at 1.2.0.0 or 1.1.0.0? Should be at the latter.

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 3]
```diff
@@ -0,0 +1,859 @@
...
+            }
+          ],
+          "minor": [],
+          "patch": []
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
+          "minor": "0",
+          "patch": "0"
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: GrandaMento is at 1.1.0.1, shouldn't there be a patch increase?

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 4]
```diff
@@ -0,0 +1,859 @@
...
+              "contract": "GrandaMento",
+              "type": "NewContract"
+            }
+          ],
+          "minor": [],
+          "patch": []
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: Ditto, does this mean 1.2 or 1.1? Should be 1.1.0.1

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 5]
```diff
@@ -0,0 +1,859 @@
...
+              "contract": "OdisPayments",
+              "type": "NewContract"
+            }
+          ],
+          "minor": [],
+          "patch": []
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: Ditto, 1.1 or 1.2? Should be 1.1.

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 6]
```diff
@@ -0,0 +1,859 @@
...
+              "contract": "StableTokenBRL",
+              "type": "NewContract"
+            }
+          ],
+          "minor": [],
+          "patch": []
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: Ditto

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 7]
```diff
@@ -0,0 +1,859 @@
...
+          "minor": "0",
+          "patch": "0"
+        }
+      },
+      "UsingRegistryV2": {
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: This shouldn't have been deployed.

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 8]
```diff
@@ -0,0 +1,859 @@
...
+          "patch": []
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
+          "minor": "0",
+          "patch": "0"
+        }
+      },
+      "UsingRegistryV2BackwardsCompatible": {
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: Ditto

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 9]
```diff
@@ -0,0 +1,859 @@
...
+              "contract": "Accounts",
+              "dependency": "Signatures",
+              "type": "LibraryLinkingChange"
+            }
+          ]
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "=",
+          "minor": "+1",
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: At the v3 tag, this was 1.1.1.1, at v8 it's 1.1.4.0, shouldn't this be a +3?

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 10]
```diff
@@ -0,0 +1,859 @@
...
+          "patch": "0"
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: This was 1.1.1.1 at v3, is 1.1.2.0 at v8, should this be a -1?

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 11]
```diff
@@ -0,0 +1,859 @@
...
+        }
+      },
+      "MetaTransactionWallet": {
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: Should this have been deployed? Not a core contract.

File: packages/protocol/releaseData/nonstandard/baklavaUpdate.json [hunk 12]
```diff
@@ -0,0 +1,7 @@
+{
+  "GrandaMento": ["0x000000000000000000000000000000000000ce10", "0x0000000000000000000000000000000000000000", "300000000000000000000000", "0", "1209600"],
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandard/baklavaUpdate.json]: Should we have set some approver for testing purposes? And some spread?

File: packages/protocol/releaseData/nonstandard/baklavaUpdate.json [hunk 13]
```diff
@@ -0,0 +1,7 @@
...
+  "StableTokenBRL": ["Celo Brazilian Real", "cREAL", 18, "0x000000000000000000000000000000000000ce10", "1000000000000000000000000", "47304000", [], [], "ExchangeBRL"],
+  "ExchangeBRL": ["0x000000000000000000000000000000000000ce10", "0x0000000000000000000000000000000000000000", "5000000000000000000000", "250000000000000000000", "300", "5"],
```
Comment (m-chrzan) [inline packages/protocol/releaseData/nonstandard/baklavaUpdate.json]: Should the stable token identifier be set to StableTokenBRL?

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json (hunk 7 above)
Comment (martinvol) [reply packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: agree but it the past it has been to alfajores and mainnet, so if I wanted to replicate I'd just have to push it

**Comment by martinvol** [comment]
Pasting the output of check versions here

```
% yarn check-versions -a core-contracts.v3 -b core-contracts.v8 -r "report_origianl_baklava.json"
yarn run v1.22.17
$ ./scripts/bash/check-versions.sh -a core-contracts.v3 -b core-contracts.v8 -r report_origianl_baklava.json
 - Checkout contracts source code at core-contracts.v3
 - Contract artifacts already built at build/core-contracts.v3
 - Checkout contracts source code at core-contracts.v8
 - Contract artifacts already built at build/core-contracts.v8
 - Checkout migrationsConfig.js at core-contracts.v8
$ /Users/martinvol/celo/celo-monorepo/node_modules/.bin/ts-node scripts/check-backward.ts sem_check --old_contracts build/core-contracts.v3/contracts --new_contracts build/core-contracts.v8/contracts --exclude '.*Test|Mock.*|I[A-Z].*|.*Proxy|MultiSig.*|ReleaseGold|SlasherUtil|UsingPrecompiles' --output_file report_origianl_baklava.json
Running storage report...Done
Running code report...Done
Running library linking...Done
Generating backward report...Done
Writing compatibility report to report_origianl_baklava.json ...Done
Version mismatch detected:
{
    "contracts": {
        "Accounts": {
            "oldVersion": {
                "storage": 1,
                "major": 1,
                "minor": 1,
                "patch": 1
            },
            "newVersion": {
                "storage": 1,
                "major": 1,
                "minor": 4,
                "patch": 0
            },
            "expectedDelta": {
                "storage": "=",
                "major": "=",
                "minor": "+1",
                "patch": "0"
            }
        },
        "Election": {
            "oldVersion": {
                "storage": 1,
                "major": 1,
                "minor": 1,
                "patch": 0
            },
            "newVersion": {
                "storage": 1,
                "major": 1,
                "minor": 2,
                "patch": 1
            },
            "expectedDelta": {
                "storage": "=",
                "major": "=",
                "minor": "+1",
                "patch": "0"
            }
        },
        "ExchangeBRL": {
            "oldVersion": null,
            "newVersion": {
                "storage": 1,
                "major": 2,
                "minor": 0,
                "patch": 0
            },
            "expectedDelta": {
                "storage": "=",
                "major": "+1",
                "minor": "0",
                "patch": "0"
            }
        },
        "Governance": {
            "oldVersion": {
                "storage": 1,
                "major": 2,
                "minor": 0,
                "patch": 2
            },
            "newVersion": {
                "storage": 1,
                "major": 2,
                "minor": 1,
                "patch": 1
            },
            "expectedDelta": {
                "storage": "=",
                "major": "=",
                "minor": "+1",
                "patch": "0"
            }
        },
        "GrandaMento": {
            "oldVersion": null,
            "newVersion": {
                "storage": 1,
                "major": 1,
                "minor": 0,
                "patch": 1
            },
            "expectedDelta": {
                "storage": "=",
                "major": "+1",
                "minor": "0",
                "patch": "0"
            }
        },
        "MetaTransactionWallet": {
            "oldVersion": {
                "storage": 1,
                "major": 1,
                "minor": 0,
                "patch": 1
            },
            "newVersion": {
                "storage": 1,
                "major": 1,
                "minor": 1,
                "patch": 1
            },
            "expectedDelta": {
                "storage": "=",
                "major": "=",
                "minor": "+1",
                "patch": "0"
            }
        },
        "Reserve": {
            "oldVersion": {
                "storage": 1,
                "major": 1,
                "minor": 2,
                "patch": 0
            },
            "newVersion": {
                "storage": 1,
                "major": 1,
                "minor": 2,
                "patch": 2
            },
            "expectedDelta": {
                "storage": "=",
                "major": "=",
                "minor": "=",
                "patch": "+1"
            }
        },
        "Validators": {
            "oldVersion": {
                "storage": 1,
                "major": 2,
                "minor": 0,
                "patch": 0
            },
            "newVersion": {
                "storage": 1,
                "major": 2,
                "minor": 0,
                "patch": 3
            },
            "expectedDelta": {
                "storage": "=",
                "major": "=",
                "minor": "=",
                "patch": "+1"
            }
        }
    }
}
```

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json (hunk 1 above)
Comment (martinvol) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]:
```suggestion
          "major": "+1",
```

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json (hunk 3 above)
Comment (martinvol) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]:
```suggestion
          "major": "0",
          "minor": "0",
          "patch": "+1"
```

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json [hunk 14]
```diff
@@ -0,0 +1,859 @@
...
+            }
+          ],
+          "minor": [],
+          "patch": []
+        },
+        "versionDelta": {
+          "storage": "=",
+          "major": "+1",
+          "minor": "0",
+          "patch": "0"
```
Comment (martinvol) [inline packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]:
```suggestion
          "patch": "+1"
```

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json (hunk 10 above)
Comment (martinvol) [reply packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: I think the tooling assumes the less significant version numbers go to zero when there's an update in the more significant

File: packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json (hunk 11 above)
Comment (martinvol) [reply packages/protocol/releaseData/nonstandarVersionReports/baklava_fix_report_report.json]: It is in the previous reports unfortunately.

File: packages/protocol/releaseData/nonstandard/baklavaUpdate.json (hunk 12 above)
Comment (martinvol) [reply packages/protocol/releaseData/nonstandard/baklavaUpdate.json]: Added the approver, spread is set as zero in mainnet.

File: packages/protocol/releaseData/nonstandard/baklavaUpdate.json (hunk 13 above)
Comment (martinvol) [reply packages/protocol/releaseData/nonstandard/baklavaUpdate.json]: it's actually the full name in mainnet for all cStables

File: packages/protocol/releaseData/nonstandard/baklavaUpdate.json [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{
+  "GrandaMento": ["0x000000000000000000000000000000000000ce10", "0x0Cc59Ed03B3e763c02d54D695FFE353055f1502D", "300000000000000000000000", "0", "1209600"],
```
Comment (martinvol) [inline packages/protocol/releaseData/nonstandard/baklavaUpdate.json:2]: Approver is baklava validator-0

**Review by pahor167 — APPROVED** [review]

**Pull request merged**